### PR TITLE
Feature/12 playing cards to playarea + Feature/95 seperation of ui and backend

### LINKED ISSUE #12: Playing cards to playarea
# User Story
As a player, I want to be able to move cards from my hand to the playarea and activate their effects during gameplay.

## Description
When a card is played from the hand, it is placed in the playarea. Nominal effects like attack and money are used at this point. The playarea should display all cards that are currently in play and their effects.

## Tasks:
- [x] #105
- [x] #27
- [x] #28
- [x] #29

### PR DESCRIPTION
**Pull request by Benjamin-Dippold**
closes #12 #95 #53 
replaces and closes #106 

# Summary
This Branch completes the Seperation of UI and backend and the setup of all nececery UI Views and their Presenters. All Card-related presenters are derived from `CardPilePresenter.java` which offers `addCardToView(CardView card)` and `removeCardFromView(CardView card)` to be used for the Client-side Message Handling. All presenters are instantiated in the MainActivity and saved as public variables, though wether they will stay there, or be instantiated elsewhere is undecided by the team. The CardView class offers the `OnClickListener()` that can facilitate sending messages to the server. A `Log.i()` containing the needed information to process the message has been left as an example.

# Regarding the high number of smells
 not all of them concern this branch and some are not big enugh to prevent this crucial merge and hold up development:

- 26x - Smells from server classes that got attributed to this branch
- 8x - Access to Presenters in MainActivity, whoose fate (where they will be created and how they will be called) is not decided yet
- 6x - TODOs that cannot/should not be finished before merging (unnececery holdup)
- 14x - Smells for commeted out code, or leftover variables which all will still be used in the TODOs.

also I have not been approved to edit Smells in SonarCloud, so I was not able document this information there.

# Regarding the low coverage
not all of them concern this branch, recalculating the coverage without Server-classes gives:

### Missed Lines:

- MainActivity: 54/54 missing Lines, as discussed the fate of MainActivity is undecided, since the Server and Client Threads might not survive switching Activities.
- Views: 34/180 missing Lines, all of them in the OnPressed() methode of CardView, since simulating a press is harder then expected and worth leaving for after the merge.
- CardPilePresenter: 3/9 missing Lines, CardIDs are not yet implemented on develop, so findViewByCardID couldn't be finished

Lines without Server/Client Classes: 641‬
Uncovered Lines without Server/Client Classes: 102
Recalculated Coverage without Server/Client Classes: 84%

# Regarding the 4 Bugs:
3 are in Server/Client classes
1 is missing CardIDs but will not hinder development

### CODE REVIEW

**Review by sosaidi — DISMISSED** [review]
The branch appears to have completed the necessary tasks and set up the UI and backend separation, UI views, and presenters. Although there are some code smells and low coverage, they are not significant enough to prevent merging. The branch appears to be in a good state for merging. LGTM!

**Review by Kranosos — APPROVED** [review]
As this is a crucial branch for the further progression of the project it is necessary to merge it in spite of it apparently not meeting the quality gate requirements. Although it is important to consider, that this can not attributed merely to this branch since itself would meet the expectations as described by Mr. Dippold in his detailed summary on this Pull-Request….
TLDR LGTM!

**Review by sosaidi — APPROVED** [review]

**Comment by Ne0Strix** [comment]
As was mentioned before me: LGTM
